erase-install.sh

```diff
@@ -620,7 +620,7 @@ confirm() {
             jh_title="${!dialog_reinstall_title}"
             jh_desc="${!dialog_reinstall_confirmation_desc}"
         fi
-        "$jamfHelper" -windowType utility -title "$jh_title" -alignHeading center -alignDescription natural -description "$jh_desc" -lockHUD -icon "$dialog_confirmation_icon" -button1 "${!dialog_cancel_button}" -button2 "${!dialog_confirmation_button}" -defaultButton 1 -cancelButton 1 2> /dev/null
+        "$jamfHelper" -windowType utility -title "$jh_title" -alignHeading center -alignDescription natural -description "$jh_desc" -lockHUD -icon "$dialog_confirmation_icon" -button1 "${!dialog_confirmation_button}" -button2 "${!dialog_cancel_button}" -defaultButton 0 -cancelButton 2 2> /dev/null
```

Comment:
If I understand your change properly, you have changed the default button from the Cancel button to the Confirm button. I don't think we should do this.

Reply:
No worries, this does only change the order of the two buttons, it does not change the default action.

But it slightly clarifies the behaviour. The old version sets button 1 as default as well as cancel. This apparently does not work in jamfHelper, as

- the Cancel button gets rendered in a strange grey color (like it's clicked)
- hitting the Enter key does nothing, instead of causing the default button to be selected. Only the ESC key works correctly for canceling the dialog.

So we explitly set the default button to 0 (i.e. there is no default button) and only set the cancel button. Functionally the same, but it is rendered correctly.

You can compare the two versions by pasting this into Terminal:
```
jamfHelper="/Library/Application Support/JAMF/bin/jamfHelper.app/Contents/MacOS/jamfHelper"
dialog_confirmation_icon="/System/Library/CoreServices/CoreTypes.bundle/Contents/Resources/AlertStopIcon.icns"
jh_title="Erasing macOS"
jh_desc="Preparing the installer may take up to 30 minutes. Once completed your computer will reboot and continue the reinstallation."
dialog_confirmation_button_en="Confirm"
dialog_confirmation_button="dialog_confirmation_button_en"
dialog_cancel_button_en="Stop"
dialog_cancel_button="dialog_cancel_button_en"
# old dialog
"$jamfHelper" -windowType utility -title "$jh_title" -alignHeading center -alignDescription natural -description "$jh_desc" -lockHUD -icon "$dialog_confirmation_icon" -button1 "${!dialog_cancel_button}" -button2 "${!dialog_confirmation_button}" -defaultButton 1 -cancelButton 1 2> /dev/null
# new dialog
"$jamfHelper" -windowType utility -title "$jh_title" -alignHeading center -alignDescription natural -description "$jh_desc" -lockHUD -icon "$dialog_confirmation_icon" -button1 "${!dialog_confirmation_button}" -button2 "${!dialog_cancel_button}" -defaultButton 0 -cancelButton 2 2> /dev/null 
```

Of course we could also remove the Cancel button and only set the Default button. But as we have to choose one version, i think having a dedicated Cancel button is more intuitive: pressing ESC to cancel the upgrade makes more sense than to press Enter which usually means "ok, i don't really care, just go on…".

Reply:
OK, I guess I didn't read it correctly. I'll let it in and generate the packages and give it a go.